### PR TITLE
Standardize mapping fields on Aggregate and Process as methods

### PR DESCRIPTION
**Pull request by qejk**

### CODE REVIEW

File: source/server/domain/aggregate.coffee [hunk 1]
```diff
@@ -28,7 +28,8 @@ class Space.eventSourcing.Aggregate extends Space.Object
 
   @registerSnapshotType: (id) ->
     fields = {}
-    fields[field] = type for field, type of this::fields
+    fields[field] = type for field, type of this::fields()
+    console.log(fields)
```
Comment (rhyslbw) [inline source/server/domain/aggregate.coffee]: remove

**Comment by rhyslbw** [comment]
@DominikGuzei and @darko-mijic I've review this and agree this is a good change. Will reassign, but this has my blessing

**Comment by DominikGuzei** [comment]
Looks great :+1: 
Ok so this is not yet ready to merge? -> **WIP** flag!

**Comment by DominikGuzei** [comment]
Ok after some discussion we decided that the update for `fields` api will go into the next major release. This needs more time and careful implementation :wink:

**Pull request closed without merging**